File: Object Oriented Programming/learning/classes_and_objects/exercises/cpp/ex-7.cpp
/*
Write a program to print the area of a rectangle 
by creating a class named 'Area' taking the values of 
its length and breadth as parameters of its constructor and 
having a function named 'returnArea' which returns the area of the rectangle. 
Length and breadth of the rectangle are entered through keyboard.
*/
#include <iostream>
using namespace std;

// define class
class Area {
    private:
        double length, breadth;

    public:
        Area(double len, double brth):
        length(len), breadth(brth) {}

        double returnArea() {
            return length * breadth;
        }
};

int main() {
    // initialize variables
    double length, breadth;

    // receive values
    cout<<"Enter Length: ";
    cin>>length;
    cout<<"Enter Breadth: ";
    cin>>breadth;

    // create rectangle object of area
    Area rectangle(length, breadth);

    // calculate and display area
    cout<<"Area of rectangle: "<<rectangle.returnArea();

    return 0;
}

File: Object Oriented Programming/learning/classes_and_objects/exercises/py/ex-7.py
"""
Write a program to print the area of a rectangle 
by creating a class named 'Area' taking the values of 
its length and breadth as parameters of its constructor and 
having a function named 'returnArea' which returns the area of the rectangle. 
Length and breadth of the rectangle are entered through keyboard.
"""

class Area:
    # initialize constructor
    def __init__(
            self,
            len = 0.0,
            brth = 0.0
    ):
        self.length = len
        self.breadth = brth
    
    def returnArea(self):
        return self.length*self.breadth
    
# initialize variable and input values
length = float(input("Enter length: "))
breadth = float(input("Enter breadth: "))

# create object of class
rectangle = Area(len=length, brth=breadth)

print(f"Area of rectangle: {rectangle.returnArea()}")




"""
### Data Encapsulation and Hiding
class Area:
    def __init(self, length=0.0, breadth=0.0):
        self.__length = length
        self.__breadth = breadth

    def setDimensions(self, length, breadth):
        self.__length = length
        self.__breadth = breadth

    def getArea(self):
        return self.__length * self.__breadth

# Initialize variable and input values
length = float(input("Enter length: "))
breadth = float(input("Enter breadth: "))

# Create an object of the class
rectangle = Area(length, breadth)

print(f"Area of rectangle: {rectangle.getArea()}")
"""